lib/note-revisions/index.tsx

```diff
@@ -3,70 +3,179 @@ import { connect } from 'react-redux';
 
 import NotePreview from '../components/note-preview';
 import TagChip from '../components/tag-chip';
+import RevisionSelector from '../revision-selector';
 import { noteCanonicalTags } from '../state/selectors';
+import isEmailTag from '../utils/is-email-tag';
 import { tagHashOf } from '../utils/tag-hash';
 
 import type * as S from '../state';
 import type * as T from '../types';
 
-type OwnProps = {
-  noteId: T.EntityId;
-  note?: T.Note;
-};
-
 type StateProps = {
-  tags: Array<{ name: T.TagName; deleted: boolean }>;
+  allTags: Map<T.TagHash, T.Tag>;
+  note: T.Note | undefined;
+  noteCanonicalTags: T.TagName[];
   noteId: T.EntityId | null;
-  note: T.Note | null;
+  revision: T.Note | undefined;
+  revisionIndex: number;
+  revisions: Map<number, T.Note> | undefined;
 };
 
-type Props = OwnProps &
-  StateProps &
+type DispatchProps = {
+  cancelRevision: () => any;
+  openRevision: (noteId: T.EntityId, version: number) => any;
+  restoreRevision: (noteId: T.EntityId, note: T.Note) => any;
+};
+
+type Props = StateProps &
+  DispatchProps &
   Pick<React.HTMLProps<HTMLDivElement>, 'aria-hidden'>;
 
-export class NoteRevisions extends Component<Props> {
+type State = {
+  includeDeletedTags: boolean;
+};
+
+export class NoteRevisions extends Component<Props, State> {
   static displayName = 'NoteRevisions';
 
+  state: State = {
+    includeDeletedTags: true,
+  };
+
+  getTags() {
+    return this.props.noteCanonicalTags
+      .map((tagName) => {
+        const tagHash = tagHashOf(tagName);
+        return { name: tagName, deleted: !this.props.allTags.has(tagHash) };
+      })
+      .filter((tag) => {
+        return this.state.includeDeletedTags || !tag.deleted;
+      });
+  }
```

Comment:
These tags were calculated in the `getRevision` selector, now since we have to use the component's state `includeDeletedTags` for the calculation, we have to place it within the component.